groovy-40: Add 4.0.25

iigs opened the pull request:
Add groovy 4.0 branch to prepare to replace groovy 2.4 branch.  Build recipe based on method used in groovy-24.

iigs committed:
groovy-40: Add 4.0.25

AndWac commented on components/developer/groovy-40/Makefile:
```diff
@@ -0,0 +1,53 @@
+COMPONENT_CLASSIFICATION=	Development/Java
+COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_PROJECT_URL=	http://groovy-lang.org/
+COMPONENT_ARCHIVE_HASH=	\
```
Please change this to https

AndWac commented on components/developer/groovy-40/Makefile:
```diff
@@ -0,0 +1,53 @@
+REQUIRED_PACKAGES += developer/java/openjdk8
+REQUIRED_PACKAGES += runtime/java/openjdk17
+REQUIRED_PACKAGES += runtime/java/openjdk21
+REQUIRED_PACKAGES += runtime/java/openjdk8
```
Are really all these jdks needed?

iigs commented:
Fixed https URL.
Identified flaw in .p5m file that included every JDK. Removed depencies on /usr/bin/java and /usr/bin/javac . Changed to manually specify one JDK package in Makefile. 
Updated license list.  Generated a new file which include all of the product's licenses.

mtelka commented on components/developer/groovy-40/Makefile:
```diff
@@ -0,0 +1,52 @@
+COMPONENT_POST_INSTALL_ACTION = $(FIND) $(PROTO_DIR) -name \*.bat -delete
+
+# JDK does not get picked up by gmake REQUIRED_PACKAGES
+REQUIRED_PACKAGES += runtime/java/openjdk21
```
You do not need to depend explicitly on the JDK.  The build framework by design makes sure it is always installed.

iigs replied:
If this was a build time dependency, it would have been explicitly stated.  No action needed.
